docs/ref/applications.txt

```diff
@@ -509,3 +514,26 @@ Here are some common problems that you may encounter during initialization:
   :setting:`INSTALLED_APPS` to contain
   ``'django.contrib.admin.apps.SimpleAdminConfig'`` instead of
   ``'django.contrib.admin'``.
+
+* ``RuntimeWarning: Accessing the database during app initialization is
+  discouraged.`` This happens if one of your apps attempts database access in
+  either the :meth:`AppConfig.ready` method or during import time. This is bad
+  because it will happen during the startup of every management command.
+
+  For example, a common mistake may be to populate the choices of a form field
+  like this::
+
+    class SomeForm(forms.Form):
+        country = forms.ChoiceField(choices=[c.name for c in Country.objects.all()])
+
+  In the example above, the ``Country.objects.all()`` query would be executed
+  as soon as the module containing this form would be imported. To silence the
+  warning, use a :class:`~django.forms.ModelChoiceField` instead::
```

Comment:
"silence the warning" typically implies ignoring it - instead, we're avoiding/fixing it. Attempted wording improvement:

```suggestion
  In the example above, the query from ``Country.objects.all()`` is executed
  when the module imports, because the ``QuerySet`` is iterated over. To avoid
  the warning, the form could use a :class:`~django.forms.ModelChoiceField`
  instead::
```